File: src/SIContentService.Contract/Models/SIContentServiceError.cs
namespace SIContentService.Contract.Models;

/// <summary>
/// Defines a SIContentService error.
/// </summary>
public sealed class SIContentServiceError
{
    /// <summary>
    /// Error code.
    /// </summary>
    public WellKnownSIContentServiceErrorCode ErrorCode { get; set; }
}

File: web/src/models/SIContentServiceError.ts
import WellKnownSIContentServiceErrorCode from "./WellKnownSIContentServiceErrorCode";

/** Defines a SIContentService error. */
export default class SIContentServiceError extends Error {
    /** Error code. */
    errorCode?: WellKnownSIContentServiceErrorCode;

    /** Error status code. */
    statusCode: number;

    constructor(message: string | undefined, statusCode: number, errorCode?: WellKnownSIContentServiceErrorCode) {
        super(message);
        this.statusCode = statusCode;
        this.errorCode = errorCode;
    }
}
